src/Swarm/Game/Scenario/Objective/Graph.hs

```diff
@@ -0,0 +1,139 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Swarm.Game.Scenario.Objective.Graph where
```

Comment:
While I applaud the attempt to have modules smaller, the lack of module documentation makes it unclear what their scope should be.

When I want to add some new function, where should I put it?

Adding a note here would clarify the intent and make it easier to maintain in the future.

Reply:
Added some descriptions.